src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Subscriptions/DependenciesSnapshotProvider.cs

```diff
@@ -160,15 +177,18 @@ public Task OnProjectFactoryCompletedAsync()
 
             Task AddInitialSubscriptionsAsync()
             {
-                // This host object subscribes to configured project evaluation data for its own purposes.
-                SubscribeToConfiguredProjectEvaluation(
-                    _activeConfiguredProjectSubscriptionService,
-                    OnActiveConfiguredProjectEvaluatedAsync);
-
-                // Each of the host's subscribers are initialized.
-                return Task.WhenAll(
-                    Subscribers.Select(
-                        subscriber => subscriber.InitializeSubscriberAsync(this, _activeConfiguredProjectSubscriptionService)));
+                lock (_lock)
```

Comment:
This is a little worrying previously it wouldn't call 3rd parties under a lock now it does. What is this protecting against?

Reply:
`IDependencyCrossTargetSubscriber` is internal and not an extension point for 3rd party code. It has two implementations, neither of which do anything particularly heavyweight. This lock is scoped to the unconfigured project so I expect contention to be low.

The lock here synchronises calls to `AddSubscriptions` with `ReleaseSubscriptions`. Given this is an init path, it may be possible to remove it. I'll investigate.

Reply:
Looking at this again that lock can be removed. It was intended to protect the initialisation of subscribers (which can happen via one of many pathways) but that protection occurs in the `Subscribers` property directly anyway, so this longer-lived lock is removed in 9e5c22ea7d39f84e706a368d38a66f79df8625c3.